### PR TITLE
feature(Documents): Expire non required documents

### PR DESCRIPTION
**Pull request by santostiago**
[Task](https://www.pivotaltracker.com/story/show/183467740).

This PR adds "not-required" documents to the list of documents to expire.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
Only valid documents get expired

Issue Number: 183467740

## What is the new behavior?
Not required documents also get expired.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app/models/operator_document.rb [hunk 1]
```diff
@@ -80,14 +80,18 @@ class OperatorDocument < ApplicationRecord
...
+    .where(status: EXPIRABLE_STATUSES)
+    .where(required_operator_documents: { contract_signature: false })
 }
 
   enum status: { doc_not_provided: 0, doc_pending: 1, doc_invalid: 2, doc_valid: 3, doc_expired: 4, doc_not_required: 5 }
   enum uploaded_by: { operator: 1, monitor: 2, admin: 3, other: 4 }
   enum source: { company: 1, forest_atlas: 2, other_source: 3 }
 
   NON_HISTORICAL_ATTRIBUTES = %w[id attachment updated_at created_at].freeze
+  EXPIRABLE_STATUSES = statuses.slice(:doc_valid, :doc_expired).values.freeze
```
Comment (tsubik) [inline app/models/operator_document.rb:94]: `%w[doc_valid doc_expired]` or `%i[doc_valid doc_expired]` would be enough and it should work I think. Rails itself should change those values to integer.

**Review by tsubik — COMMENTED** [review]
You can add another spec, pretty simple for `#expire_documents` in `operator_document_spec.rb` just to add another document with different status

**Review by tsubik — APPROVED** [review]
Other than missing spec, I think all good. Expiration task should that we already have running daily should expire those not required documents.

**Comment by santostiago** [comment]
@tsubik I updated the PR to include tests.
I also realized we were not expiring gov documents, so I added them to the job (I'll confirm it with Sophie, but I think we weren't expiring them by mistake, and not by design).

**Comment by santostiago** [comment]
Also, we should (later) refactor the `gov documents` and `operator documents` to include a `documents` module. Like 50% of the code is the same.

File: app/admin/required_gov_document.rb [hunk 2]
```diff
@@ -20,12 +20,11 @@
       rd.deleted_at.nil?
     end
     column 'required_gov_document_group' do |rd|
-      rd.required_operator_document_group&.name
+      rd.required_gov_document_group&.name
```
Comment (tsubik) [inline app/admin/required_gov_document.rb:23]: :+1:

File: spec/factories/gov_documents.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FactoryBot.define do
+  factory :gov_document, class: GovDocument do
+    required_gov_document
+    expire_date { Date.tomorrow }
+    start_date { Date.yesterday }
+    current { true }
+
+    transient do
+      force_status { nil }
```
Comment (tsubik) [inline spec/factories/gov_documents.rb:9]: Wouldn't work if we just set status? Anyway, interesting use of transient.

**Review by tsubik — APPROVED** [review]
Looks great. New specs are very well organized. I'm always struggling with that ;)

File: spec/factories/gov_documents.rb (hunk 3 above)
Comment (santostiago) [reply spec/factories/gov_documents.rb:9]: I always do it like this. I'm not sure if `status` would work.

**Pull request merged**